setup.py

```diff
@@ -1,5 +1,6 @@
 import os
 import platform
+import multiprocessing
```

Comment:
not sure this speeds up compiling, it's mostly cmake that runs in parallel already?

Reply:
My bad. I did not want to commit that (It's supposed to be an RFC on another PR). I'll revert it.

it does speed things up things significantly (for me at least, on macOs). when you call cmake build it does not run in parallel by default. The problem is that this command is only available in a slightly more recent CMake version than the one we currently require.

Reply:
OK cool, then we should upgrade the required CMake version

Reply:
Currently we require cmake `3.10.3`. `--parallel` requires `3.12`. Current version is `3.16`.
According to cmake's website `3.10.3` was released on March 18, 2018,  while `3.12` is dated July 17, 2018. So it's just 4 months more recent.

Reply:
I'll bump the required version on `CMakeLists.txt` then?